### DISCUSSION TITLE
More ONNX models available?

### DISCUSSION THREAD

**Discussion by JakubPelka** [PUTvision/qgis-plugin-deepness#161] [General]
Hi, 
I am more a GIS person than AI-computer vision person. 
I tried your tool and it is great ! But... 
can you create link somewhere that will point where one can find more model to use directly in plugin? 
one with the cars - works great, but... where do I find more similar for segmentation, upscaling etc. 

It will help a lot to all GIS-people who want to use your plugin in daily job.
Will be glad for that!

**Comment by przemyslaw-aszkowski** [comment]
Hi,

Thanks for your feedback! All the models we are aware of are listed in the Model ZOO:
https://qgis-plugin-deepness.readthedocs.io/en/latest/main/main_model_zoo.html

Unfortunately, creating an AI model is not so simple, as it requires a lot of training data.
Once you have a dataset, models are quite simple to train and use in Deepness.
If you, or anyone else, have some interesting ideas and can provide a dataset, we would be happy to train new models.
For a detection model (like the one for cars), you need at least a few hundred detections in the training dataset. 

Cheers
Przemek

**Comment by JakubPelka** [comment]
Hi and thanks for answer!
Well - data is not a big issue :) but if you can create a simple step-by-step manual for GIS people and not CV people how to prepare it to make it work it will be really appreciated. 
We can stay in touch about it! :) 

I was also wondering - if you can add rotated bounding-box solution so it will fit for ex. cars better ?

**Reply by curtis18** [reply]
Hello @przemyslaw-aszkowski , I would also appreciate it if you could provide the steps for creating the model for deepness. I tried to create a model by Yolo, and converted the onnx format. Unfortunately, it cannot detect anything using deepness. But if using yolo to detect the image, the output is normal.

**Reply by przemyslaw-aszkowski** [reply]
Hi,
For the segmentation models there is an example training provided here: https://qgis-plugin-deepness.readthedocs.io/en/latest/example/example_segmentation_landcover.html

For detection, there is a training example in Python code: https://qgis-plugin-deepness.readthedocs.io/en/latest/example/example_detection_cars_yolov7.html

And another instruction provided by one of the user for training without a single line of code:
https://qgis-plugin-deepness.readthedocs.io/en/latest/creators/creators_without_line_of_code.html

Let me know if it is helpful or if there are any issues
Cheers
Przemek

**Reply by jducnuigeen** [reply]
Hi [JakubPelka](https://github.com/JakubPelka), I'm currently working through training my first instance segmentation model on Google Colab with my own imagery stored in Google Drive, and I too am just a GIS end user. Once I get a model working in Deepness, I plan to write up my entire workflow and share with others. Hopefully that will be helpful.

**Comment by JakubPelka** [comment]
Well exactly Jan D..
It would be great to ask a GIS - person to write that manual :D
Because I guess it is easy for ones with CV skills :)


tis 8 okt. 2024 kl. 21:38 skrev Jan Ducnuigeen ***@***.***>:
